us.springett.cvss.MalformedVectorException comes with 3.2.2

CvssV2.class
![image](https://github.com/vaadin/appsec-kit/assets/18628242/3aed6e18-fd65-49c2-9e70-f23c02da3dff)
![image](https://github.com/vaadin/appsec-kit/assets/18628242/379cf09e-97e5-4d88-a8a6-0386ad2e671e)

Caused by: us.springett.cvss.MalformedVectorException: Unknown metric: CVSS
	at us.springett.cvss.CvssV2$Parser.parseVector(CvssV2.java:321)
	at us.springett.cvss.CvssV2$Parser.parseVector(CvssV2.java:246)
	at us.springett.cvss.Cvss.fromVector(Cvss.java:51)
	at com.vaadin.appsec.backend.AppSecDTOProvider.lambda$getHighestCvssScoreNumber$5(AppSecDTOProvider.java:354)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.reduce(ReferencePipeline.java:662)
	at java.base/java.util.stream.ReferencePipeline.max(ReferencePipeline.java:698)
	at com.vaadin.appsec.backend.AppSecDTOProvider.getHighestCvssScoreNumber(AppSecDTOProvider.java:357)
	at com.vaadin.appsec.backend.AppSecDTOProvider.findSeverityIfHigher(AppSecDTOProvider.java:333)
	at com.vaadin.appsec.backend.AppSecDTOProvider.updateVulnerabilityStatistics(AppSecDTOProvider.java:309)
	at com.vaadin.appsec.backend.AppSecDTOProvider.lambda$getDependencies$3(AppSecDTOProvider.java:104)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)
	at com.vaadin.appsec.backend.AppSecDTOProvider.getDependencies(AppSecDTOProvider.java:107)
	at com.vaadin.appsec.backend.AppSecService.getDependencies(AppSecService.java:308)
	at com.vaadin.appsec.views.DependenciesView.configureSearchField(DependenciesView.java:189)
	at com.vaadin.appsec.views.DependenciesView.<init>(DependenciesView.java:70)
	at com.vaadin.appsec.views.AppSecView.buildTabSheet(AppSecView.java:124)
	at com.vaadin.appsec.views.AppSecView.buildLayout(AppSecView.java:75)
	at com.vaadin.appsec.views.AppSecView.<init>(AppSecView.java:59)

--- a/appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/AppSecDTOProvider.java
+++ b/appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/AppSecDTOProvider.java
@@ -351,6 +351,7 @@ class AppSecDTOProvider {
     private Double getHighestCvssScoreNumber(
             OpenSourceVulnerability vulnerability) {
         return vulnerability.getSeverity().stream()
+                .filter(severity -> isSupportedCvssType(severity.getType()))
                 .map(severity -> Cvss.fromVector(severity.getScore()))
                 .filter(Objects::nonNull)
                 .map(cvss -> cvss.calculateScore().getBaseScore())
@@ -367,18 +368,24 @@ class AppSecDTOProvider {
         String cvssString = "";
         double tempBaseScore = 0.0;
         for (Severity severity : vulnerability.getSeverity()) {
-            Cvss cvss = Cvss.fromVector(severity.getScore());
-            if (cvss != null) {
-                double baseScore = cvss.calculateScore().getBaseScore();
-                if (baseScore > tempBaseScore) {
-                    tempBaseScore = baseScore;
-                    cvssString = severity.getScore();
+            if (isSupportedCvssType(severity.getType())) {
+                Cvss cvss = Cvss.fromVector(severity.getScore());
+                if (cvss != null) {
+                    double baseScore = cvss.calculateScore().getBaseScore();
+                    if (baseScore > tempBaseScore) {
+                        tempBaseScore = baseScore;
+                        cvssString = severity.getScore();
+                    }
                 }
             }
         }
 
         return tempBaseScore >= highestScoreNumber ? cvssString
                 : highestScoreString;
+    }
+
+    private boolean isSupportedCvssType(Severity.Type type) {
+        return type == Severity.Type.CVSS_V2 || type == Severity.Type.CVSS_V3;
     }
 
     private Optional<String> getPatchedVersion(Affected affected) {

--- a/appsec-kit-backend/src/test/java/com/vaadin/appsec/backend/AppSecDTOProviderTest.java
+++ b/appsec-kit-backend/src/test/java/com/vaadin/appsec/backend/AppSecDTOProviderTest.java
@@ -30,6 +30,7 @@ import com.vaadin.appsec.backend.model.osv.response.OpenSourceVulnerability;
 import com.vaadin.appsec.backend.model.osv.response.Package;
 import com.vaadin.appsec.backend.model.osv.response.Range;
 import com.vaadin.appsec.backend.model.osv.response.Reference;
+import com.vaadin.appsec.backend.model.osv.response.Severity;
 
 import static org.mockito.ArgumentMatchers.anyList;
 import static org.mockito.Mockito.mock;
@@ -158,15 +159,20 @@ public class AppSecDTOProviderTest {
                     }
                 });
 
+        Severity severity1 = new Severity(Severity.Type.CVSS_V3,
+                "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:N/A:N");
+        Severity severity2 = new Severity(Severity.Type.CVSS_V4,
+                "CVSS:4.0/AV:L/AC:L/AT:N/PR:L/UI:N/VC:H/VI:N/VA:N/SC:N/SI:N/SA:N");
+
         OpenSourceVulnerability vulnerability1 = createVulnerability(
                 "GHSA-mjmj-j48q-9wg2", "CVE-2022-1471", reference,
-                List.of(affected1));
+                List.of(affected1), List.of(severity1, severity2));
         OpenSourceVulnerability vulnerability2 = createVulnerability(
                 "GHSA-9j49-mfvp-vmhm", "CVE-2021-23406", reference,
-                List.of(affected2, affected3));
+                List.of(affected2, affected3), List.of());
         OpenSourceVulnerability vulnerability3 = createVulnerability(
                 "GHSA-9j49-mfvp-vmhm", "CVE-2021-23406", reference,
-                List.of(affected2, affected3));
+                List.of(affected2, affected3), List.of());
 
         return Arrays.asList(vulnerability1, vulnerability2, vulnerability3);
     }
@@ -194,21 +200,23 @@ public class AppSecDTOProviderTest {
 
         OpenSourceVulnerability vulnerability1 = createVulnerability(
                 "GHSA-mjmj-j48q-9wg2", "CVE-2022-1471", reference,
-                List.of(affected1));
+                List.of(affected1), List.of());
         OpenSourceVulnerability vulnerability2 = createVulnerability(
                 "GHSA-9j49-mfvp-vmhm", "CVE-2021-23406", reference,
-                List.of(affected2));
+                List.of(affected2), List.of());
 
         return Arrays.asList(vulnerability1, vulnerability2);
     }
 
     private OpenSourceVulnerability createVulnerability(String id, String alias,
-            Reference reference, List<Affected> affected) {
+            Reference reference, List<Affected> affected,
+            List<Severity> severity) {
         OpenSourceVulnerability vulnerability = new OpenSourceVulnerability();
         vulnerability.setId(id);
         vulnerability.setAliases(Collections.singletonList(alias));
         vulnerability.setReferences(Collections.singletonList(reference));
         vulnerability.setAffected(affected);
+        vulnerability.setSeverity(severity);
         return vulnerability;
     }
 

